packages/jest-jasmine2/src/index.js

```diff
@@ -69,7 +69,32 @@ function jasmine2(config, environment, moduleLoader, testPath) {
       moduleLoader.requireModule(null, config.setupTestFrameworkScriptFile);
     }
   });
+
+  const iterableEquality = (left, right) => {
+    let result; // result needs to be `undefined` if not an iterator
+    if (left && right && left[Symbol.iterator] && right[Symbol.iterator]) {
+      const leftIterator = left[Symbol.iterator]();
+      const rightIterator = right[Symbol.iterator]();
+      let nextLeft = leftIterator.next();
+      let nextRight = rightIterator.next();
+
+      result = true;
+      while (result && !nextLeft.done) {
+        result = jasmine.matchersUtil.equals(nextLeft, nextRight);
```

Comment:
Since we know the values in the iterator object upfront, it seems easier to me to do this:

```js
if (
  !jasmine.matchersUtil.equals(aItem.value, bItem.value) ||
  aItem.done !== bItem.done
) {
```